routers/init.go

```diff
@@ -42,6 +43,24 @@ func NewServices() {
 	cache.NewContext()
 }
 
+// In case of problems connecting to DB, retry connection. Eg, PGSQL in Docker Container on Synology
+func initDBEngine() (err error) {
+	log.Info("Beginning ORM engine initialization.")
+	for i := 0; i < setting.DBConnectRetries; i++ {
+		log.Info("ORM engine initialization attempt #%d/%d...", i+1, setting.DBConnectRetries)
+		if err := models.NewEngine(migrations.Migrate); err == nil {
```

Comment:
Here should have been:

```go
if err = ...
```

The reason the compiler didn't alarm is because there was a named err always available. I knew I never liked that feature of the language!

Reply:
ur right.  I did have it that way the first go around, then I changed it due to a compile error (which was for another unrelated problem).  I currently have em on separate lines tho.  u want me to condense it back down?

Reply:
Push your current changes up, and let's see.

Reply:
I pushed it already.   It didn't show up here I guess becz its merged?

Reply:
Should I hit the "compare & new PR" button. u reckon?

Reply:
Yup

Reply:
well i went ahead and condensed it back down to a one-liner, sans the re-declaration